activerecord/lib/active_record/connection_configurations.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require "active_record/connection_configurations/connection_config"
+
+module ActiveRecord
+  class ConnectionConfigurations # :nodoc:
```

Comment:
I'm not too fond of calling this internal concept "connection" / "connection configuration"... given `AR::Base#connection` and friends, that feels like it adds avoidable ambiguity. (And e.g. `AR::Base.configurations` contains the config for database connections, but `AR::Base.connection_configurations` contains a different [but related] thing.)

Perhaps, as we've extended our definition of a DB connection a la `configurations` (first through read-only conns, and now shards), we need to explicitly acknowledge [e.g.] "database" as a separate first-class noun [to which you might have both readonly and read-write connections], as well as invent another one ("data store"?) which is the aggregate of several shards.